src/Raven.Server/Platform/Posix/LimitsReader.cs

```diff
@@ -0,0 +1,279 @@
+                LimitsInfo.Current.MapCountCurrent = await GetCurrentMapCountAsync();
+                LimitsInfo.Current.ThreadsCurrent = GetCurrentThreadsCount();
+                LimitsInfo.Current.SetValues();
+            }
+            finally
+            {
+                _lock.Release();
+            }
+
+            return LimitsInfo.Current;
+        }
+
+        public static LimitsInfo ReadMaxLimits()
+        {
+            if (PlatformDetails.RunningOnPosix == false)
+                throw new InvalidOperationException("Cannot read Current Limits because it requires POSIX");
+
+            if (_lock.Wait(0) == false)
+                return LimitsInfo.Current;
+
+            try
+            {
+                LimitsInfo.Current.MapCountMax = GetMaxMapCount();
+                LimitsInfo.Current.ThreadsMax = GetMaxThreadsCount();
+                LimitsInfo.Current.SetValues();
+            }
+            finally
+            {
+                _lock.Release();
+            }
+
+            return LimitsInfo.Current;
+        }
+
+        private static long GetMaxValueForLimit(string limit)
+        {
+            var maxValueString = File.ReadAllText(limit);
+            if (long.TryParse(maxValueString, out var maxValueLong) && maxValueLong > 0)
+            {
+                return maxValueLong;
+            }
+
+            throw new InvalidOperationException($"Could not parse the value of '{limit}', got: '{maxValueString}'.");
+        }
+
+        private static long GetMaxThreadsCount()
+        {
+            return GetMaxValueForLimit(ThreadsMaxFilePath);
+        }
+
+        private static long GetMaxMapCount()
+        {
+            return GetMaxValueForLimit(MaxMapCountFilePath);
+        }
+
+        private static long GetCurrentThreadsCount()
+        {
+            var arrString = File.ReadAllText(ThreadsCurrentFilePath);
+            var arr = arrString.Split(' ');
+
+            var maxValueString = arr[19]; // num_threads  %ld
+
+            if (long.TryParse(maxValueString, out var maxValueLong) && maxValueLong > 0)
+            {
+                return maxValueLong;
+            }
+
+            throw new InvalidOperationException($"Could not parse the value of '{ThreadsCurrentFilePath}', got: '{maxValueString}'.");
+        }
+
+        private static async Task<long> GetCurrentMapCountAsync()
+        {
+            long currentMapCount;
+            await using (FileStream stream = File.OpenRead(CurrentMapCountFilePath))
+            {
+                currentMapCount = await GetEolCountAsync(stream);
+            }
+
+            return currentMapCount;
+        }
+
+        private static async Task<long> GetEolCountAsync(Stream stream)
+        {
+            var pipe = new Pipe();
+            var writing = FillPipeAsync(stream, pipe.Writer);
+            var reading = ReadPipeAsync(pipe.Reader);
+            await Task.WhenAll(reading, writing);
+
+            return await reading;
+        }
```

Comment:
This is way too complex code.
Use something like:

```
using var f = File.OpenRead(CurrentMapCountFilePath);
var reader = PipeReader.Create(f);
int totalLines = 0;
while (true)
{
    var read = await reader.ReadAsync();
    if (read.IsCompleted)
        break;

    var buf = read.Buffer;
    while (true)
    {
        var pos = buf.PositionOf((byte)'\n');
        if (pos == null)
            break;
        totalLines++;
        buf = buf.Slice(buf.GetPosition(1, pos.Value));
    }
    reader.AdvanceTo(buf.Start, buf.End);
}
```

Reply:
done